jobs/ml_jobs/item_linkage/link_items.py

```diff
@@ -59,16 +51,11 @@ def get_links(
     Returns:
         pd.DataFrame: Dataframe containing matched pairs.
     """
-    # TODO: chainer
     features = comparator.compute(candidate_links, table_left, table_right)
     matches = features[features.sum(axis=1) >= MATCHES_REQUIRED].reset_index()
     matches = matches.rename(columns={"level_0": "index_1", "level_1": "index_2"})
-    # matches.assign(
-    #     index_1=lambda df: f"L-{df['index_1']}",
-    #     index_2=lambda df: f"R-{df['index_2']}",
-    # )
-    matches["index_1"] = matches["index_1"].apply(lambda x: f"source-{x}")
-    matches["index_2"] = matches["index_2"].apply(lambda x: f"candidate-{x}")
+    matches["index_1"] = matches["index_1"].apply(lambda x: f"S-{x}")
+    matches["index_2"] = matches["index_2"].apply(lambda x: f"C-{x}")
```

Comment:
pourquoi ne pas laisser source ? C'est plus explicite que S et C non ?

Reply:
en pandas, ça sera beaucoup plus rapide d'utiliser des opérations "vectorisées" , plutôt que des boucles for. Et en gros, les map, apply sont des boucles for déguisées. 

par exemple ici tu peux faire :

```
matches = matches.assign(
  index_1 = lambda df: "S-" + df.index_1.astype(str),
  index_2 = lambda df: "C-" + df.index_2.astype(str)
)
```

Pour savoir si tu peux vectoriser ou non, il faut se poser la question; est-ce que mon opération va dépendre de la valeur d'un champ de ma ligne, ou est ce que je peux appliquer une opération globale à une colonne ? (genre ici, on concatène une colonne avec un string => c'est donc vectorisable)